### PR TITLE
Add coverage test for CPU only and variable batch size test

### PR DESCRIPTION
**Pull request by JanuszL**
- adds a test that checks if all available DALI operators are covered
  by CPU only and variable batch size test. The test lists all operators
  and checks if the list of tested operators matches it. The list is
  manually maintained and need to be updated when a new test is added

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It adds coverage test for CPU only and variable batch size test

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     adds a test that checks if all available DALI operators are covered by CPU only and variable batch size test. The test lists all operators and checks if the list of tested operators matches it. The list is manually maintained and need to be updated when a new test is added
 - Affected modules and functionalities:
     test_dali_variable_batch_size.py
     test_dali_cpu_only.py
 - Key points relevant for the review:
     check list correctness
 - Validation and testing:
     CI run
 - Documentation (including examples):
     NA

**JIRA TASK**: *[DALI-2032]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

File: dali/test/python/test_dali_cpu_only.py [hunk 1]
```diff
@@ -817,4 +818,159 @@ def test_separated_exec_setup():
...
+    "dl_tensor_python_function",
+    "hidden.arithmetic_generic_op", #internal
+    "hidden.transform_translation", #internal
+    "video_reader",         # not supported for CPU
+    "video_reader_resize",  # not supported for CPU
+    "readers.video",        # not supported for CPU
+    "readers.video_resize", # not supported for CPU
+    "optical_flow",         # not supported for CPU
+]
+def test_coverage():
```
Comment (klecki) [inline dali/test/python/test_dali_cpu_only.py:957]: Please put it in some test utils and reuse in both cases.

File: dali/test/python/test_dali_cpu_only.py [hunk 2]
```diff
@@ -817,4 +818,159 @@ def test_separated_exec_setup():
...
+            if len(prefix):
+                prefix += '.'
+            else:
+                prefix = ""
+            res.append(prefix + cls.__name__)
+        else:
+            for c in cls.__dict__.keys():
+                if not c.startswith("_") and c not in sys.builtin_module_names:
+                    c = cls.__dict__[c]
+                    res += get_functions(c, cls.__name__)
```
Comment (klecki) [inline dali/test/python/test_dali_cpu_only.py:972]: Dunno how it works, but when I imported the `fn` module, and did this:
`fn.__dict__['readers'].__dict__['file'].__name__`, it is `'file'` and not `'nvidia.dali.fn.readers.file'` or `'readers.file'`. So I'm thinking there might be a problem with the prefix handling and not reporting submodules of `fn` as `readers.file` for example.

**Review by klecki — COMMENTED** [review]
As a general remark:
we are not able to test plugin operators that we include by default (pytorch, numba?).

There arises a question if we should test all the arithmetic/mathematical ops. (And the `math` module). Now it's only a subset, with no detection.

File: dali/test/python/test_dali_cpu_only.py (hunk 2 above)
Comment (JanuszL) [reply dali/test/python/test_dali_cpu_only.py:972]: That is why you have prefix argument. In your case you get `fn.__dict__['readers'].__name__` as prefix, then you strip `nvidia.dali.fn` and prepend to `fn.__dict__['readers'].__dict__['file'].__name__`.

File: dali/test/python/test_dali_cpu_only.py (hunk 1 above)
Comment (JanuszL) [reply dali/test/python/test_dali_cpu_only.py:992]: Done

**Comment by JanuszL** [comment]
> As a general remark:
> we are not able to test plugin operators that we include by default (pytorch, numba?).

Not automatically. At least I don't have a good idea how to extract only relevant info from plugins modules where we have much more than we need.

> There arises a question if we should test all the arithmetic/mathematical ops. (And the `math` module). Now it's only a subset, with no detection.

Added math module.

**Review by klecki — APPROVED** [review]

File: dali/test/python/test_dali_variable_batch_size.py [hunk 3]
```diff
@@ -692,3 +697,187 @@ def pipe(max_batch_size, input_data, device, input_layout):
...
+    "math.fpow",
+    "math.acosh",
+    "math.min",
+    "pytorch.TorchPythonFunction",
+    "numba.fn.experimental.numba_function",
+    "hidden.transform_translation", # intentional
+    "hidden.arithmetic_generic_op", # intentional
+    "coco_reader",              # readers are do not support variable batch size yet
+    "sequence_reader",          # readers are do not support variable batch size yet
+    "numpy_reader",             # readers are do not support variable batch size yet
```
Comment (szalpal) [inline dali/test/python/test_dali_variable_batch_size.py:854]:
```suggestion
    "numpy_reader",             # readers do not support variable batch size yet
```

**Review by szalpal — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

File: dali/test/python/test_dali_variable_batch_size.py (hunk 3 above)
Comment (JanuszL) [reply dali/test/python/test_dali_variable_batch_size.py]: Done

**Pull request merged**